src/main/java/org/traccar/protocol/RuptelaProtocolDecoder.java

```diff
@@ -236,7 +236,18 @@ protected Object decode(
                 buf.readUnsignedByte(); // timestamp extension
 
                 if (type == MSG_EXTENDED_RECORDS) {
-                    buf.readUnsignedByte(); // record extension
+                    int b = buf.readUnsignedByte(); // record extension
+                    int noRecordsToMerge = (b & 0xf0) >> 4;
+                    int currentRecord = b & 0x0f;
+
+                    if (currentRecord > 0 && noRecordsToMerge >= currentRecord) {
+                        if (!positions.isEmpty()
+                                && positions.get(positions.size() - 1).getDeviceTime()
+                                   .compareTo(position.getDeviceTime()) == 0) {
```

Comment:
Why do we need this comparison?

Reply:
It's an additional safety check, that we are truly merging two Positions with the same time (as all such extended Positions have the same header, so time, lat/long,...

Reply:
I would remove any unnecessary checks.